packages/block-library/src/list-item/edit.js

```diff
@@ -1,47 +1,259 @@
+/**
+ * External dependencies
+ */
+import { first } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import {
 	RichText,
 	useBlockProps,
 	useInnerBlocksProps,
+	BlockControls,
+	store as blockEditorStore,
 } from '@wordpress/block-editor';
-import { __ } from '@wordpress/i18n';
+import { isRTL, __ } from '@wordpress/i18n';
 import { createBlock } from '@wordpress/blocks';
+import { useDispatch, useSelect, select } from '@wordpress/data';
+import { ToolbarButton } from '@wordpress/components';
+import {
+	formatOutdent,
+	formatOutdentRTL,
+	formatIndentRTL,
+	formatIndent,
+} from '@wordpress/icons';
+import { useCallback } from '@wordpress/element';
+
+function createListItem( listItemAttributes, listAttributes, children ) {
+	return createBlock(
+		'core/list-item',
+		listItemAttributes,
+		! children || ! children.length
+			? []
+			: [ createBlock( 'core/list', listAttributes, children ) ]
+	);
+}
+
+function useIndentListItem( clientId ) {
+	const { canIndent } = useSelect(
+		( innerSelect ) => {
+			const { getBlockIndex } = innerSelect( blockEditorStore );
+			return {
+				canIndent: getBlockIndex( clientId ) > 0,
+			};
+		},
+		[ clientId ]
+	);
+	const { replaceBlocks, selectionChange } = useDispatch( blockEditorStore );
+	return [
+		canIndent,
+		useCallback( () => {
+			const {
+				getBlockRootClientId,
+				getBlock,
+				getBlockOrder,
+				getSelectionStart,
+				getSelectionEnd,
+				getBlockIndex,
+			} = select( blockEditorStore );
```

Comment:
I thought we discouraged the usage of global `select`. Should we use the new selector getting method here?

```js
const {
  getBlockRootClientId,
  getBlock,
  getBlockOrder,
  getSelectionStart,
  getSelectionEnd,
  getBlockIndex,
} = useSelect( blockEditorStore );
```

Reply:
Good catch, yeah, seems like we should fix that.

Reply:
I'll create a follow-up PR 👍

Reply:
Done #39821.